### DISCUSSION TITLE
DB4S Crashes When Apple Machine in Clamshell Mode with External Monitor

### DISCUSSION THREAD

**Discussion by alexbastuba** [sqlitebrowser/sqlitebrowser#3343] [General]
Hello,
 
I encountered a bug today that I thought would be worthwhile mentioning to the devs of this wonderful tool.

I am new to SQL and have been going through an essentials course that recommended DB4S. This is an amazing tool, thank you for creating it!

My Macbook Pro 2021 ARM M1 Pro machine works flawlessly with this application with the built-in monitor, however if I am using clamshell mode (laptop closed, external power, HDMI connection to external monitor) DB browser crashes with a SIGABORT error. Unfortunately the error logs are no longer popping up, I will edit if they appear again.

This error only occurs when in clamshell mode and only using my external monitor. If my laptop is open while connected to the external monitor, I can drag DB4S onto the external monitor screen space with no issues. If I open DB4S in this way and then close my laptop to enter clamshell mode, DB4S also crashes. 

This is not a major issue as I have found a workaround by keeping the laptop open while using my external monitor, but it seemed critical enough to mention to the devs if it has not been reported before.

Thank you,

Alex

**Comment by justinclift** [comment]
Interesting.  That's when using our latest release (3.12.2) yeah?

That being the case, would you be ok to try it with our very latest nightly build?

https://nightlies.sqlitebrowser.org/latest/

That'll let us know whether the problem still exists with our current development code or not. :smile:

**Reply by lucydodo** [reply]
Haha, I'm a step behine. 😄

**Reply by alexbastuba** [reply]
I also want to mention that I installed the public build through homebrew so I uninstalled and reinstalled, and also uninstalled and reinstalled through the normal .dmg drag to applications folder method. Both of those still resulted in this crash. I'll try the nightly build now!

**Reply by lucydodo** [reply]
You can also install nightly version with following command: `brew install --cask db-browser-for-sqlite-nightly`

**Reply by alexbastuba** [reply]
`brew` cannot find the nightly build:
Warning: Cask 'db-browser-for-sqlite-nightly' is unavailable: No Cask with this name exists.

**Reply by lucydodo** [reply]
Sorry, Before installing, you'll also need the following commands: `brew tap homebrew/cask-versions`

**Reply by alexbastuba** [reply]
Okay, thank you for that tip! I didn't know that.

I installed the 3.12.99 version through the .dmg link above. 

It still crashed, but here is the error log:
[ClamshellCrashError3.12.99.txt](https://github.com/sqlitebrowser/sqlitebrowser/files/11468928/ClamshellCrashError3.12.99.txt)

**Reply by lucydodo** [reply]
Thanks. After analyzing the report, it looks like the problem is with the `NSOpenGLContext` or `NSOpenGLView` class.
I've seen a few similar cases on the internet, some users were able to fix it by clearing PRAM/SMC, and 
some users say it's a macOS Monterey-specific issue.

Anyway, upgrading a macOS is a bit complicated. Why not try a PRAM/SMC clearing first? 😄 
Here's Apple's official documentation on the subject:
- PRAM: https://support.apple.com/en-us/HT201255
- SMC: https://support.apple.com/en-us/HT201295

**Reply by lucydodo** [reply]
To summarize Apple's documentation:
- PRAM reset: Press Option-Command-P-R when you start Mac. (On a black screen immediately after turn on)
> It will automatically reboot when it's resetd (You'll probably hear startup sound two times)
- SMC reset: Turn off Mac, and then press power button for 10 secs, and then start Mac.

**Reply by alexbastuba** [reply]
Okay, I'll try both of those resets and report back. If that doesn't work, I've been meaning to upgrade to Ventura and this would be a good reason to get that done. 

Thanks for your time and analyzing the error log, that's wild you were able to pick those details out from it. 

Edit: It looks like the M1 Macs don't have a traditional PRAM and SMC reset anymore, I shut down my machine for 30+ seconds which supposedly is as close as you can get to a reset but it's still crashing. I not willing to go about updating to Ventura tonight, but when I do in the near future I'll make a new comment on this post. Thank you for your time and I'm sorry there wasn't an immediate solution.

**Reply by lucydodo** [reply]
Oh, I'm really sorry. I just upgraded from an Intel Mac to an M2 Mac and had no idea.
I'll also install macOS Monterey on my secondary Mac and give it a try, Thanks for the report this issue. :smile:

**Comment by lucydodo** [comment]
It's weird. on my secondary Mac it works fine in clamshell mode. Of course, that could be because my secondary Mac is so old.
> (MacBook Pro 2012 with macOS Catalina)

Have you tried the nightly version by any chance?
You can download latest nightlies version the following: https://nightlies.sqlitebrowser.org/latest/

**Reply by justinclift** [reply]
What's a "sub-Mac"?  I don't know that term. :smile:

**Reply by lucydodo** [reply]
Hmm, I meant a secondary Mac.
That's how we say it in my country, but it's probably bad English. I learned one, Thank you! 😄

**Reply by justinclift** [reply]
Ahhh.  Cool, all good.

Thinking about how I'd say the same thing, I'd probably say something like "my other mac".  That's just my first thought anyway. :smile:

**Reply by lucydodo** [reply]
"my other mac" good, I'll try that next time I have something like this 😄

**Comment by FerFerreyra90** [comment]
Hi everyone! 

Exploring DB Browser, its great! I'm starting browsing on DataBases.

I'm having the same issue but not in Clamshell Mode. Crashed several times, I updated Mac OS and SMC Reset No Help.
Attahced Crash Log
[DBCrashErrortxt.txt](https://github.com/sqlitebrowser/sqlitebrowser/files/11473377/DBCrashErrortxt.txt)
Thanks you all :)

Fernando

**Reply by chrisjlocke** [reply]
And did you try the latest nightly version, as per above?

**Reply by FerFerreyra90** [reply]
I did! Now its not crashing!
Thanks!!!

**Reply by chrisjlocke** [reply]
Excellent, thanks for confirming.

Weirdly, the master is a codebase of 'fixes' from nightly, but I'm wondering if there is a deeper underlying issue, as the nightly solves 98.32% of weird crashes.

**Reply by FerFerreyra90** [reply]
Hello Folks!

My DB Browser is crashing again, I downloaded and installed the late nightly but its not working....
[Crash.txt](https://github.com/sqlitebrowser/sqlitebrowser/files/11630784/Crash.txt)

thanks you all again

**Reply by chrisjlocke** [reply]
From your post from two weeks ago, it was not crashing.  If you install that version and it does crash, it seems unlikely its the software at fault, as it hasn't changed.  If the version from two weeks ago does not crash but the latest version is crashing, then that's more interesting.

So for the avoidance of doubt, if you install the version from two weeks ago which ran fine for you, is that version now causing a problem on your system?
